engine/src/main/java/io/camunda/zeebe/process/test/engine/GrpcToLogStreamGateway.java

```diff
@@ -642,4 +680,46 @@ private DecisionEvaluationRecord createDecisionEvaluationRecord(
   public String getAddress() {
     return "0.0.0.0:" + port;
   }
+
+  private final class GrpcJobConsumer implements JobConsumer, AutoCloseable {
+    private final DirectBuffer jobType;
+    private final ServerCallStreamObserver<ActivatedJob> observer;
+    private final Executor executor;
+
+    private GrpcJobConsumer(
+        final DirectBuffer jobType, final ServerCallStreamObserver<ActivatedJob> observer) {
+      this.jobType = jobType;
+      this.observer = observer;
+      executor = MoreExecutors.newSequentialExecutor(ForkJoinPool.commonPool());
```

Comment:
:thought_balloon: gRPC docs requires us to synchronize access to the observer, so we use a sequential executor here wrapping the common fork join pool. Locks were an alternative to avoid async code, but I wasn't sure if that would impact the in memory engine since I'm not familiar with how it's written.